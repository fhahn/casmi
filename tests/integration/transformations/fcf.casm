// transform, cmdline "--fcf"

CASM fcf

// constant folding

init initR

function af : -> Int
function bf : -> Int

rule initR = 
{
	bf := BVadd_result(32, 1, 1)
	af := 2*3 + 5

    program(self) := undef
}
