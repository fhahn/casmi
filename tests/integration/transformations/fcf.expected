initR=STATEMENTS(LET CONST(ident c0__,num 2),STATEMENTS(LET CONST(ident c1__,num 11),STATEMENTS(LET CONST(ident c2__,SELF),STATEMENTS(LET CONST(ident c3__,UNDEF),PARBLOCK(STATEMENTS(UPDATE(ident bf,ident c0__),STATEMENTS(UPDATE(ident af,ident c1__),STATEMENTS(UPDATE(LOOKUP(ident program,LIST(ident c2__,null)),ident c3__),null))),null)))))